website/docs/r/msk_cluster.html.markdown

```diff
@@ -187,11 +187,21 @@ The following arguments are supported:
 * `instance_type` - (Required) Specify the instance type to use for the kafka brokersE.g., kafka.m5.large. ([Pricing info](https://aws.amazon.com/msk/pricing/))
 * `security_groups` - (Required) A list of the security groups to associate with the elastic network interfaces to control who can communicate with the cluster.
 * `az_distribution` - (Optional) The distribution of broker nodes across availability zones ([documentation](https://docs.aws.amazon.com/msk/1.0/apireference/clusters.html#clusters-model-brokerazdistribution)). Currently the only valid value is `DEFAULT`.
+* `connectivity_info` - (Optional) Information about the cluster access configuration. See below. For security reasons, you can't turn on public access while creating an MSK cluster. However, you can update an existing cluster to make it publicly accessible. You can also create a new cluster and then update it to make it publicly accessible ([documentation](https://docs.aws.amazon.com/msk/latest/developerguide/public-access.html)).
+
+### broker_node_group_info connectivity_info Argument Reference
+
+* `public_access` - (Optional) Access control settings for brokers. See below.
+
+### connectivity_info public_access Argument Reference
+
+* `type` - (Optional) Public access type. Valida values: `DISABLED`, `SERVICE_PROVIDED_EIPS`.
 
 ### client_authentication Argument Reference
 
 * `sasl` - (Optional) Configuration block for specifying SASL client authentication. See below.
 * `tls` - (Optional) Configuration block for specifying TLS client authentication. See below.
+* `unauthenticated` - (Optional) Enables unauthenticated access.
```

Comment:
@ewbankkit Does this default to false? Just a heads up, but prior to this option being introduced clusters would default to it enabled when plaintext options were specified.

The AWS API call in this case does not return the key at all but its implicitly enabled.
I think in later revisions of the API it forces it to be specified if those options are enabled else it reports and error. SOo probably OK?

I was just worried about the cases where it was implicitly enabled but the provider then defaults it to false if the property is absent, which is why i added a validate method. But can see the reasons for removing this as it replcaes logic in the AWS API.

I's been a while, but see this comment:
https://github.com/hashicorp/terraform-provider-aws/issues/20956#issuecomment-929116934

Reply:
Actually, we are testing that scenario in the new `TestAccKafkaCluster_ClientAuthenticationTLS_initiallyNoAuthentication` acceptance test where we start with no `client_authentication` and `encryption_in_transit.client_broker = "PLAINTEXT"` and then ensure that we can update in-place to multiple `client_authentication` methods and `encryption_in_transit.client_broker = "TLS"`.